src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -218,77 +221,92 @@ private void visitEachToken(DetailAST node) {
+     * @param codePoints the array of Unicode code points
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the character at {@code columnNo} is a space, and
      *         not preceded by another space.
      */
-    private static boolean isSingleSpace(String line, int columnNo) {
-        return isSpace(line, columnNo) && !Character.isWhitespace(line.charAt(columnNo - 1));
+    private static boolean isSingleSpace(int[] codePoints, int columnNo) {
+        return isSpace(codePoints, columnNo) && !isWhitespace(codePoints, columnNo - 1);
     }
 
     /**
      * Checks if the {@code line} at {@code columnNo} is a space.
      *
-     * @param line The line in the file to examine.
+     * @param codePoints the array of Unicode code points
      * @param columnNo The column position in the {@code line} to examine.
      * @return {@code true} if the character at {@code columnNo} is a space.
      */
-    private static boolean isSpace(String line, int columnNo) {
-        return line.charAt(columnNo) == ' ';
+    private static boolean isSpace(int[] codePoints, int columnNo) {
+        return codePoints[columnNo] == ' ';
     }
 
     /**
-     * Checks if the {@code line} at {@code columnNo} is a whitespace character.
+     * Checks if the {@code line} up to and including {@code columnNo} is all
+     * non-whitespace text encountered.
      *
-     * @param line The line in the file to examine.
+     * @param codePoints the array of Unicode code points
      * @param columnNo The column position in the {@code line} to examine.
-     * @return {@code true} if the character at {@code columnNo} is a
-     *         whitespace.
+     * @return {@code true} if the column position is the first non-whitespace
+     *         text on the {@code line}.
      */
-    private static boolean isWhitespace(String line, int columnNo) {
-        return Character.isWhitespace(line.charAt(columnNo));
+    private static boolean isFirstInLine(int[] codePoints, int columnNo) {
+        return  codePoints.length == 0 || isNonWhitespace(codePoints, columnNo - 1);
     }
 
     /**
-     * Checks if the {@code line} up to and including {@code columnNo} is all
-     * non-whitespace text encountered.
+     * Checks if the codePoints is empty,
+     * or contains only whitespace characters.
      *
-     * @param line The line in the file to examine.
-     * @param columnNo The column position in the {@code line} to examine.
-     * @return {@code true} if the column position is the first non-whitespace
-     *         text on the {@code line}.
+     * @param codePoints the array of Unicode code points
+     * @index till index to check
+     * @return true if the arg is blank.
      */
-    private static boolean isFirstInLine(String line, int columnNo) {
-        return CommonUtil.isBlank(line.substring(0, columnNo));
+    public static boolean isNonWhitespace(int[] codePoints, int index) {
```

Comment:
It might be a good idea just to make an overloaded `CommonUtil#indexOfNonWhitespace(int[] codePoints)`, then you can just check that return value of method  == column number in `isFirstInLine()` on line 271 above. This will likely help with coverage as well.

Reply:
Or, create a `CommonUtil#indexOfCharacter(int[] codePoints, char character)` ? Then you could also use this method in `GenericWhitespaceCheck` to find `&`.